hypothesis-python/tests/cover/test_observability.py

```diff
@@ -63,6 +68,7 @@ def do_it_all(l, a, x, data):
 
 
 @xfail_on_crosshair(Why.other, strict=False)  # flakey BackendCannotProceed ??
+@skipif_threading  # captures observations from other threads
```

Comment:
Hmm, could we capture to a `dict[thread_id, list[observation]]` or otherwise make this work?  I'd certainly like observability to function even under multithreading, but the current approach will likely have problems with concurrent writes...

Reply:
I think the former makes sense; each test has a single responsible thread etc.  The catch is that our default callback is "append to a file", which is not itself especially threadsafe - we might need to insert a queue or something.

Reply:
I'll address this in a separate PR